image-js/image-js
# dsstore

- chore: tabula rasa
- lerna init
- install some dev dependencies and create core package
- add typescript config
- add poc for tests
- setup testing with multiple packages
- add travis config
- implement image constructor and pixel methods
- add values iterator and implement invert
- make data public and add tests
- apply prettier
- simplify core to one package
- add getOutputImage and allow to construct only with options
- fix TS error
- remove 32-bit support for now
- rename bitDepth to depth
- add eslint
- mark the library as side-effect-free
- prepare for ESM and rollup builds
- add png decoder
- test: allow to import with aliases
- add decode
- fix test not being executed
- no need to build tests
- fix eslint
- f
- improve coverage
- slightly better check header
- add typedoc
- add some documentation to decode functions
- decode jpeg
- auto decode jpeg
- fix moduleResolution TS config
- make the ESM build
- disable npm ci during bootstrap
- chore: update fast-png to TS version
- encode png
- simplify createFrom and make rollup work
- add wip demo folder
- working demo
- add readImg, fill, fillChannel
- update encode png test
- add fillAlpha method
- add read method
- test: Image tests back to 100% coverage
- test: perfect coverage in validators
- implement some of the color conversions
- eslint fix
- add rgb(a) conversion
- put color convert hot path in function for optimization
- test error if converting to same kind
- add failing jpeg encode test
- remove test-travis call
- add webcam demo
- add rgb to grey
- convert color before writing to canvas
- extract webcam script and add selector
- chore(package): update fast-png to version 3.1.3 (#446)
- update dependencies
- fix jpeg decode
- chore: update fast-png to version 4
- add invert tests and remove bad copy option from getOutputImage
- add depth conversion
- update deps
- add image.split()
- fix lerna compilation
- fix tsconfig
- add polyfill for array.flat in tests
- add interpolateBorder method
- fix bug in convertDepth and add tests
- remove test image
- Update README.md
- Update README.md
- move writeCanvas to save directory
- add encode, write and simplify types
- fix getImage and add getTestImage
- add histogram computation
- work on write and encode with tests
- add support for thresholding
- fix tsc and add roi package
- make threshold value or algorithm mandatory
- add draft of roisFromMask
- wip: convolution
- fix typings of convolution
- fix tsconfig
- fix test tsconfig
- prepare a publish-canary script
- fix convolution to match opencv
- remove unused image
- separate round and clamp
- chore: lint all ts files
- add test for roisFromMask
- test: make sure everything is build before running tests
- fix: make sure path aliases are only available in tests
- v1.0.0-alpha.0
- add separated kernels normalization
- do not export normalizeSeparatedKernel
- add direct 2d convolution
- refactor: better variable and function names in convolution
- fix typo
- add test for normalized separated kernel
- remove comment
- chore: update publish script
- fix: ThresholdAlgorithm should be a string
- remove log
- chore: update dependencies
- add licence back
- fix remaining bugs in convolution
- add test for gaussian blur
- skip failing test
- test: remove unused variable
- fix blur options
- v1.0.0-alpha.1
- feat: add readSync and writeSync
- optimize webcam demo to show bottlenecks
- feat: allow out option in convolution
- Add Greenkeeper badge 🌴 (#460)
- implement transform and add test with simple translation
- test: remove unused variable
- update ml-convolution to 2.0.0
- v1.0.0-alpha.2
- optimize convolution
- implement rotation
- fix eslint errors
- implement bilinear interpolation
- round bilinear interpolation result
- greatly simplify border interpolation
- update demo to allow arbitrary scripts
- fix rotate with default options
- implement resize method
- implement merge method
- expose methods on prototype
- test: fix rotate test
- add bicubic interpolation
- prepare test for bilinear and bicubic interpolations
- rename variables for clarity
- add custom inspect function
- rewrite resize closer to expected result
- add resize todo example
- fix(package): update type-fest to version 0.4.0 (#467)
- fix(package): update ml-matrix to version 6.0.0 (#469)
- Update fs-extra to the latest version 🚀 (#470)
- chore: update dependencies
- Bump lodash.template from 4.4.0 to 4.5.0 (#487)
- Bump mixin-deep from 1.3.1 to 1.3.2 (#489)
- Bump lodash from 4.17.11 to 4.17.15 (#490)
- Bump eslint-utils from 1.3.1 to 1.4.2 (#488)
- chore: add .DS_Store in gitignore
